queenbee/schema/arguments.py

```diff
@@ -293,7 +293,29 @@ class WorkflowArguments(Arguments):
     field for user_data.
     """
     user_data: Dict = Field(
-        None,
+        {},
+        description='Optional user data as a dictionary. User data is for user reference'
+        ' only and will not be used in the execution of the workflow.'
+    )
+
+
+class WorkflowInputs(BaseModel):
```

Comment:
What's the difference between `WorkflowInputs` and `WorkflowArguments`? Why do we need this one?

Reply:
Workflow Inputs are the payload sent to the user to modify the Workflow arguments. In your initial implementation your proposed the following schema which does not match the Worfklow arguments so I formalised it:

```
        The dictionary should have two keys for ``parameters`` and ``artifacts``.
        The values for each key is a dictionary where the key is the name of the
        target variable and value is ``key: value`` that should be updated.

        For instance this input updates the value for workers to 6 and grid-name to
        classroom.

        .. code-block:: python

            {
                'parameters': {
                    'worker': {'value': 6},
                    'grid-name': {'value': 'classroom'}
                }
            }

```